### ISSUE TITLE
Emoji push the line they're on down into the line below

### ISSUE THREAD

**Issue by lee-dohm** [zeit/hyper#154]
<img width="718" alt="screen shot 2016-07-16 at 12 08 46 am" src="https://cloud.githubusercontent.com/assets/1038121/16893340/9bae1804-4ae9-11e6-9c72-0aa8e629d9bb.png">

In this screenshot, the top two lines and the bottom two lines are evenly spaced. But the middle line with the emoji is pushed downward and collides with the line below it. You can even see it when the row is highlighted in the Inspector:

<img width="320" alt="screen shot 2016-07-16 at 12 12 00 am" src="https://cloud.githubusercontent.com/assets/1038121/16893356/faac2198-4ae9-11e6-940d-248e72fcf10d.png">

This is the output of Homebrew's `brew upgrade` command, if you want to try and replicate it.

**Comment by martinwolf** [comment]
I guess I have the same issue.

<img width="535" alt="screen shot 2016-07-16 at 10 58 11" src="https://cloud.githubusercontent.com/assets/1442939/16893832/36af4184-4b44-11e6-8ebf-bcd68100aaf1.png">

**Comment by Charlotteis** [comment]
I have the same issue. I tried increasing the padding on `x-row`s (each line in the terminal appears to be an `x-row`) but the same thing still happened.

**Comment by montogeek** [comment]
Duplicate of https://github.com/zeit/hyperterm/issues/31
Removing emojis fixes the issue :)

**Comment by Charlotteis** [comment]
I'm just wondering if there is a way to fix it and allowing us to keep emojis.
1: Homebrew uses emojis, many people have Homebrew.
2: Lots of people use emoji in the terminals now.

I am of course happy to help fix this, I just don't know how yet :D

**Comment by montogeek** [comment]
I would be more than happy if we found a way to solve it, I miss my emojis 😭

**Comment by vaff** [comment]
Same problem here. Running [Wild Cherry Theme](https://github.com/mashaal/wild-cherry) this is really a issue:

![screen shot 2016-07-18 at 11 34 48](https://cloud.githubusercontent.com/assets/351447/16910944/f5635f00-4cdb-11e6-9981-f2ae5f8700c2.png)

I tried bumping the line height to 2em. This just fucks up the cursor:
![screen shot 2016-07-18 at 11 36 25](https://cloud.githubusercontent.com/assets/351447/16910970/1193130a-4cdc-11e6-9563-3935a17fcced.png)

**Comment by ChiperSoft** [comment]
I noticed this immediately when I launched Hyperterm the first time. I have my shell prompt set to `export PS1="🏠  \w $ "` and it causes the cursor to be misaligned. I also noticed that the `$` in my prompt gets erased when I start typing.

![image](https://cloud.githubusercontent.com/assets/231157/16922501/99752414-4cca-11e6-8e1d-5473262e1769.png)

![image](https://cloud.githubusercontent.com/assets/231157/16922479/86ba868e-4cca-11e6-8694-bd296f3a1797.png)

**Comment by scottluptowski** [comment]
Reposting this from the closed #290 PR

My terminal is set up so that next to the folder name I always display three emojis.

This is how it looks in bash:

![screen shot 2016-07-19 at 11 45 25 am](https://cloud.githubusercontent.com/assets/3178471/16956512/b9f17454-4da6-11e6-84eb-518b84f3253e.png)

This is how it looks in HyperTerm:
![screen shot 2016-07-19 at 11 45 21 am](https://cloud.githubusercontent.com/assets/3178471/16956520/c556284e-4da6-11e6-9778-e5dadeb8102e.png)

And to prove this is emoji related, and not related to the length of the string, I replaced the emoji with text:
![screen shot 2016-07-19 at 11 47 04 am](https://cloud.githubusercontent.com/assets/3178471/16956532/d4a179b6-4da6-11e6-941b-4319f43ba102.png)

**Comment by shockey** [comment]
I'm seeing the same: 

![messages image 3841548251 _censored](https://cloud.githubusercontent.com/assets/680248/17969750/799bd60c-6a88-11e6-9a79-395bfffa5896.jpg)

**Comment by andersryanc** [comment]
Just add `x-row { line-height: 1em; }` to your `termCSS` config setting.

**Comment by Jonas-Drotleff** [comment]
This only happens when you use unicode characters that are not included in the font you are using.

For Example: I use Monaco so "▲" is displayed properly but any other emojis are not. When I switch to another font such as Space Mono that does not support "▲" then the lines are moved

**Comment by daveaspinall** [comment]
Adding `x-row { line-height: 1em; }` for me didn't fix the issue. I can modify the line height to work, but the cursor still overlaps the text because the emoji must be slightly wider than a standard character.

![screen shot 2016-11-18 at 09 16 49](https://cloud.githubusercontent.com/assets/681517/20424841/c6dc78d4-ad6f-11e6-8dd7-4b5da089fdeb.png)

Any ideas? Kind of rules Hyper.app out for me unless this can be resolved 😭

**Comment by m10l** [comment]
Same issue too. I use emoji cat faces to determine whether there are changes to commit on a Git repo. It's **incredibly** important that these are cat faces. **Incredibly**. 

![screen shot 2016-12-02 at 15 28 15](https://cloud.githubusercontent.com/assets/4446634/20837342/f9aa2466-b8a3-11e6-84ac-c369e218b80f.png)

**Comment by pablopunk** [comment]
@m10l offtopic, what colors/plugins do u use to get that scheme? 😁

**Comment by andersryanc** [comment]
@pablopunk looks kinda like the Atom Dark theme: https://www.npmjs.com/package/hyperterm-atom-dark

**Comment by dotcypress** [comment]
[Related PR](hyper/pull/1098)

**Comment by ChiperSoft** [comment]
Do we need to make a new issue for the problem of unicode characters being too wide and throwing off the cursor position, like in @m10l's screenshot?

**Comment by daveaspinall** [comment]
@dotcypress @matheuss Fix mentioned in #154 doesn't fix our issue. Emoji characters in prompt still knock the cursor out after pulling the latest commits?

![screen shot 2016-12-09 at 08 17 52](https://cloud.githubusercontent.com/assets/681517/21042260/4455d548-bde8-11e6-8bb8-01997c8a67c7.png)

**Comment by daveaspinall** [comment]
```
x-row::first-letter {
  margin-right: -0.5em;
}
```

This quickly fixes the emoji "width" issue but breaks lines that don't include the emoji 😕

Any way to target just the emoji (PS1)?

**Comment by daveaspinall** [comment]
```
x-row > span:first-child {
  margin-right: -0.5em;
}
```

A little hacky, but fixes the emoji width issue 👌🏼  Might add it to a pull request but needs more testing!

**Comment by m10l** [comment]
@pablopunk I'm using [hyper-dark-dracula](https://www.npmjs.com/package/hyper-dark-dracula) and modified version of the crunch zsh theme. You can grab it here -  https://gist.github.com/m10l/753940d27e44a6e6f9e0a0f27b8a9cfb

**Comment by AndrewHamm** [comment]
@ChiperSoft Have you opened a new issue? The line height fix does not fix the horizontal overlapping for me.

You can see in this photo that the cursor is to the right of where the characters appear. As I type the characters do not appear where expected.
![screen shot 2017-01-06 at 8 19 11 pm](https://cloud.githubusercontent.com/assets/7542350/21738052/902c6dee-d44d-11e6-805c-c95906acb73c.png)
![jan-06-2017 20-27-26](https://cloud.githubusercontent.com/assets/7542350/21738134/91c746f0-d44e-11e6-86e2-f1c3f2e61da8.gif)

**Comment by ChiperSoft** [comment]
My issue got corrected with the 1.0 release, so I didn't bother.
